crates/nu-command/src/filesystem/ls.rs

```diff
@@ -338,14 +338,22 @@ fn is_hidden_dir(dir: impl AsRef<Path>) -> bool {
     #[cfg(windows)]
     {
         use std::os::windows::fs::MetadataExt;
+        let hidden_attributes: bool;
 
         if let Ok(metadata) = dir.as_ref().metadata() {
             let attributes = metadata.file_attributes();
             // https://docs.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants
-            (attributes & 0x2) != 0
+            hidden_attributes = (attributes & 0x2) != 0;
         } else {
-            false
+            return false;
         }
+
+        hidden_attributes
```

Comment:
Document this here, explaining why this is necessary in windows. Most probably due to the use of Linux software that ueses .(dot) files instead of the hidden attribute.

Reply:
In Windows, every file has attributes, one of them is the hidden attribute(second bit) and this one determinate if a file is visible or not, while in Linux this is determined by their name ( .(dot)files are hidden). I change the code to check the both cases (hidden attribute and .(dot)file) before show the file, as this comment suggested:
> I think as long as it's only junctions that are hidden, not other types of symlinks - it would be fine, as Windows Explorer also hides junctions by default.
> 
> But in general, this points to a larger feature - have filesystem filters configurable in the user config. Here's why:
> 
> If the goal is to have the cleaner filesystem for everyday tasks, then dot files and underscore files also need to be hidden. Some versions of Vim use `C:\Users\user\_vimrc`, and there's a lot of dot files in my Windows user home too:
> 
> ```
> .android
> .ash_history
> .cache
> .cargo
> .config
> .gvimrc
> .InstallAnywhere
> .jeveassets
> .lesshst
> .m2
> .matplotlib
> .pyfa
> .rustup
> .ssh
> .thumbnails
> .vimrc
> .vscode
> ```
> 
> It wouldn't hurt to just quickly resolve this issue by modifying `ls` specifically for junctions, but later I'd love to see a proper configurable filter, because so far the dot files far outnumber the junctions in my home dir, they're the bigger offender.

Reply:
I mean add a comment to the code.

Reply:
The comment is incorrect, you are not checking the hidden attributes. What you need to explain (in correct English, use a website for that) that in Windows, besides the `hidden` attribute, you might want to check the file name and hide file names that start with a dot. This is due to the fact that many Linux programs, when used in Windows, do not set the `hidden` attribute, but rather use the Linux convention of the dot file.